### PR TITLE
Add error message for running Metal.jl under Rosetta

### PR DESCRIPTION
**Pull request by tgymnich**
Metal.jl does not seem to work under Rosetta (#338) so add a warning for now until we find a fix.

### CODE REVIEW

File: src/utilities.jl [hunk 1]
```diff
@@ -14,7 +14,11 @@ macro sync(code)
 end
 
 function versioninfo(io::IO=stdout)
-    println(io, "macOS $(macos_version()), Darwin $(darwin_version())")
+    print(io, "macOS $(macos_version()), Darwin $(darwin_version())")
+    if process_translated()
+        print(io, ", using Rosetta")
+    end
+    println(io)
```
Comment (christiangnrd) [inline src/utilities.jl:21]: What if you implemented the same mechanism forunsupported platforms as in [CUDA.jl](
https://github.com/JuliaGPU/CUDA.jl/blob/2a3cec926ba48f5bb4efbec5c9242259370963fa/src/initialization.jl#L9-L29)
```suggestion
    @assert functional(true)
    
    println(io, "macOS $(macos_version()), Darwin $(darwin_version())")
```

Comment (tgymnich) [reply src/utilities.jl:21]: isn't `functional()` on its way out?

```
    # TODO: make errors here (and in submodules/subpackages like cuBLAS and cuDNN) fatal,
    #       and remove functional(), once people sufficiently use weak dependencies.

```

Comment (christiangnrd) [reply src/utilities.jl:21]: In that case this looks good to me.

**Review by christiangnrd — APPROVED** [review]

**Review by maleadt — APPROVED** [review]
LGTM, thanks.

**Pull request merged**